### DISCUSSION TITLE
bpf_probe_read_kernel  usage in tcplife

### DISCUSSION THREAD

**Discussion by tahirali-csc** [iovisor/bcc#4753] [Q&A]
In `tcplife.bpf.c`, when reading ip4/pi6 source/dest addresses  https://github.com/iovisor/bcc/blob/8422cd449ad2e60414a4508aa4a150a9db39c4a3/libbpf-tools/tcplife.bpf.c#L138, we use `bpf_probe_read_kernel` instead of `bpf_core_read`. The whole purpose of  `bpf_core_read` is that handles fields offset changes transparently across different kernel version. Any thoughts?

**Comment by chenhengqi** [comment]
The relocation is handled by `BPF_CORE_READ(args, saddr)`.
